Storage/ExtensionUtils.swift

```diff
@@ -65,24 +79,35 @@ public struct ExtensionUtils {
             }
         }
 
+        // See if the text is URL-like enough to be an url, in particular, check if it has a valid TLD.
+        func textToUrl(_ text: String) -> URL? {
```

Comment:
Adds fuzzy logic to see if selection is URL-like and has a TLD, if so treat it as an URL (which provides more options in the share dialog).